classes/tdx-signed-harden.inc

```diff
@@ -9,10 +9,23 @@
 # combination UBOOT_SIGN_ENABLE="0" TDX_UBOOT_HARDENING_ENABLE="1" is not very
 # helpful and can prevent the loading of non-FIT images.
 #
-# Because of the above, we enable the hardening only if both TDX_IMX_HAB_ENABLE
-# and UBOOT_SIGN_ENABLE are set, by default.
-#
-TDX_UBOOT_HARDENING_ENABLE ?= "${@'1' if bb.utils.to_boolean(d.getVar('TDX_IMX_HAB_ENABLE')) and bb.utils.to_boolean(d.getVar('UBOOT_SIGN_ENABLE')) else '0'}"
+# Moreover, currently the hardening is tested only with Torizon OS and it's
+# not expected to work out of the box with BSP images.
+#
+# Because of the above, we enable the hardening only when building Torizon OS
+# images provided that both TDX_IMX_HAB_ENABLE and UBOOT_SIGN_ENABLE are set, by
+# default.
+#
+def default_tdx_uboot_hardening_enable(d):
+    # TODO: Remove this condition once support for BSP images is added.
+    if "torizon" not in d.getVar("OVERRIDES").split(":"):
+        return False
+    if (bb.utils.to_boolean(d.getVar('TDX_IMX_HAB_ENABLE')) and
+        bb.utils.to_boolean(d.getVar('UBOOT_SIGN_ENABLE'))):
+        return True
+    return False
+
+TDX_UBOOT_HARDENING_ENABLE ?= "${@'1' if default_tdx_uboot_hardening_enable(d) else '0'}"
```

Comment:
I suppose someone could still overwrite this variable in a conf file somewhere to be set for non-Torizon OS builds right? Should we try and print a warning in such cases or just leave the user to the consequences?

I'm fine with either honestly since the user would be making a conscious decision despite our documentation.

Reply:
Good point. I've added the check to stop build if one tries to use the unsupported configuration.

Reply:
Btw, I chose to throw a fatal error because we were already doing that for the other combination of unsupported configuration. Opinions?

Reply:
Is this to say that it will never let someone build in an unsupported configuration?
I would suggest a warning instead as it may be wanted regardless of support from toradex (e.g. custom support is being added by the implementer).

Reply:
> Btw, I chose to throw a fatal error because we were already doing that for the other combination of unsupported configuration. Opinions?

I don't have a super strong opinion on fatal error vs warning. Though I do admit that throwing a fatal error more clearly communicates our intent and expected usage of this layer, and is consistent with other cases.

> I would suggest a warning instead as it may be wanted regardless of support from toradex (e.g. custom support is being added by the implementer).

Could you elaborate? If a contributor adds support and submits a PR here then we could easily revise the error/warning handling in this case at that time. If experimenting then simply removing/commenting out the point of throwing the error would be simple as well.

Reply:
The problem would be that another implementer would not be able to build or test as the bbfatal would stop the build process.
The likely hood is that I will be looking into this in the next few weeks to implement any required changes in one of our layers.
Once tested and working, I would then apply the same changes and feed back as a PR here.
My main problem is that my main development machine is offline with a nightly update (fork) of the repos we depend on.

Reply:
> The problem would be that another implementer would not be able to build or test as the bbfatal would stop the build process.

I don't see this as a problem. The implementer in question would just change the bbfatal in their implementation then and submit that as part of the PR. This does not permanently prevent anything in the future. This is just about communicating the intent and current state of the layer as it exists now.

Reply:
Just to clarify my previous....
The build environment I work in does not allow me direct access to modify any public sources.
Ideally, I would pull the git repo, figure out what changes are needed to fix any warnings and issues, then move over to my online system and "eyeball" the changes over into a writable web-based version that I would then create a PR for.

Reply:
If there was maybe an extra "undocumented" flag that I could use to "ignore" the error, then that would also be OK. As I could then set that while developing and fixes.
Another "Hack" would be to add the checked string into OVERRIDES, but not sure that would be a good idea as not sure if that may trigger other changes in dependant layers.

Reply:
@MMG-DG 

> Another "Hack" would be to add the checked string into OVERRIDES, but not sure that would be a good idea as not sure if that may trigger other changes in dependant layers.

This should work for your use case since the only other layer depending on the "torizon" override is `meta-toradex-torizon` which you're not supposed to be using.

Reply:
@MMG-DG we appreciate a lot your feedback.

We will merge this because having a feature that is currently broken might cause more harm than good.

Be aware that this PR is just a workaround. We plan to review this soon, so it is possible that this feature will be available to BSP users soon.

For your specific used case, in case you want to take the risks, please add `torizon` to `OVERRIDES` to use the U-Boot hardening feature.